src/asm2wasm.h

```diff
@@ -725,6 +734,93 @@ class Asm2WasmBuilder {
     return call;
   }
 
+  // Some conversions might trap, so emit them safely if necessary
+  Expression* makeTrappingFloatToInt(Expression* value) {
+    if (trapMode == TrapMode::Allow) {
+      auto ret = allocator.alloc<Unary>();
+      ret->value = value;
+      ret->op = ret->value->type == f64 ? TruncSFloat64ToInt32 : TruncSFloat32ToInt32;
+      ret->type = WasmType::i32;
+      return ret;
+    }
+    // WebAssembly traps on float-to-int overflows, but asm.js wouldn't, so we must do something
+    // First, normalize input to f64
+    auto input = value;
+    if (input->type == f32) {
+      auto conv = allocator.alloc<Unary>();
+      conv->op = PromoteFloat32;
+      conv->value = input;
+      conv->type = WasmType::f64;
+      input = conv;
+    }
+    // We can handle this in one of two ways: clamping, which is fast, or JS, which
+    // is precisely like JS but in order to do that we do a slow ffi
```

Comment:
This behavior is fine for now; I'd be very interested in seeing how wasm build of `__fixsfsi` or `__fixdfdi` compares with FFI+JS version.